validator/server_api/validation_client.go

```diff
@@ -96,14 +98,66 @@ func (c *ValidationClient) Room() int {
 	return int(room32)
 }
 
+type GetLeavesHashRequest struct {
+	ExecutionID       uint64
+	FromBatch         uint64
+	MachineStartIndex uint64
+	StepSize          uint64
+	NumDesiredLeaves  uint64
+}
+
+// Generic marshaller
+type jsonMarshaller[T any] struct{}
```

Comment:
I like jsonMarshaller..
it brings up the question: should Publisher/subscriber avoid Marshaller and instead use json.Marshall / json.Unmarshal?